src/System.Private.Uri/src/System/IPv4AddressHelper.cs

```diff
@@ -27,7 +27,18 @@ internal static string ParseCanonicalName(string str, int start, int end, ref bo
             {
                 byte* numbers = stackalloc byte[NumberOfLabels];
                 isLoopback = Parse(str, numbers, start, end);
-                return numbers[0] + "." + numbers[1] + "." + numbers[2] + "." + numbers[3];
+
+                Span<char> stackSpace = stackalloc char[NumberOfLabels * 3 + 3];
+                int totalChars = 0, charsWritten;
+                for (int i = 0; i < 3; i++)
+                {
+                    numbers[i].TryFormat(stackSpace.Slice(totalChars), out charsWritten);
+                    int periodPos = totalChars + charsWritten;
+                    stackSpace[periodPos] = '.';
+                    totalChars = periodPos + 1;
+                }
```

Comment:
How is totalChars accumulating given its an out not ref parameter?

Reply:
`charsWritten` is an `out`, not `totalChars`. `totalChars` is just a local.